build.gradle.kts

```
@@ -1,214 +1,259 @@
+            .directory(workingDir)
+            .redirectOutput(ProcessBuilder.Redirect.PIPE)
+            .redirectError(ProcessBuilder.Redirect.PIPE)
+            .start()
+
+    proc.waitFor(1, TimeUnit.MINUTES)
+    return proc.inputStream.bufferedReader().readText().trim()
+}
+
+fun getSourceControlMetadata() : Triple<String, String, String> {
+    val branch: String
+    val commitHash: String
+    val qualifier: String
+    if(System.getenv("GITHUB_ACTIONS").isNullOrEmpty()) {
+        println("Running locally")
+        branch = "git rev-parse --abbrev-ref HEAD".runCommand()
+        commitHash = "git rev-parse --verify --short HEAD".runCommand()
+        qualifier = ""
+    } else {
+        println("Running under GitHub Actions")
+        branch = System.getenv("GITHUB_REF")
+        commitHash = System.getenv("GITHUB_SHA").substring(0, 7)
+        qualifier = "alpha"
+    }
+    return Triple(branch, commitHash, qualifier)
+}
+
+val sharedManifest = the<JavaPluginConvention>().manifest {
+    attributes["Signature-Version"] = version
+    attributes["Codename"] = codename
+    attributes["Build-Date"] = LocalDateTime.now()
+
+    val (branch, gitRevision, qualifier) = getSourceControlMetadata()
+    attributes["Full-Buildstring"] = "$branch@$gitRevision"
+    attributes["Git-Revision"] = gitRevision
+    attributes["Build-Qualifier"] = qualifier
+}
+
+// Main JAR
+tasks.jar {
+    dependsOn(configurations.runtimeClasspath)
+    from(sourceSets.main.get().output)
+    from({
+        configurations.runtimeClasspath.get().filter { it.name.contains("javahelp") && it.name.endsWith("jar") }.map { zipTree(it) }
+        configurations.runtimeClasspath.get().filter { it.name.contains("picocli") && it.name.endsWith("jar") }.map { zipTree(it) }
+    })
+    manifest {
+        attributes["Main-Class"] = application.mainClassName
+        from(sharedManifest)
+    }
+    dependsOn("copyHelp")
+}
+
+tasks.assemble{
+    dependsOn("jar") 
+}
+
+// NoHelp JAR
+tasks.create<Jar>("noHelpJar"){
+    classifier = "nohelp"
+    dependsOn(configurations.runtimeClasspath)
+    from(sourceSets.main.get().output)
+    from({
+        configurations.runtimeClasspath.get().filter { it.name.contains("picocli") && it.name.endsWith("jar") }.map { zipTree(it) }
+    })
+    manifest {
+        attributes["Main-Class"] = application.mainClassName
+        from(sharedManifest)
+    }
+}
+
+/*
+ * Code coverage report tasks
+ */
+tasks.jacocoTestReport {
+    reports {
+        xml.isEnabled = true
+        csv.isEnabled = false
+        html.isEnabled = false
+    }
+}
+
+tasks.check{
+    dependsOn("jacocoTestReport")
+}
+
+tasks.register("release") {
+    group = "Release"
+    description = "Creates all artifacts for a given EduMIPS64 release"
+    dependsOn("allDocs")
+    dependsOn("jar")
+
+    doFirst {
+        println("Creating artifacts for version $version")
+    }
+}
+
+tasks.create<Exec>("exeCreator"){

```

```suggestion
tasks.create<Exec>("createMsi"){

```

Since this creates an MSI installer, I suggest to call it `createMsi` or something similar.